integration_test/build-adapters.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+base_name="gcr.io/${1}/healthcare-api-dicom-dicomweb-adapter-"
+
+build_adapter () {
+  adapter_name=${base_name}${1}
+  version=$(cat build.gradle | grep "version = '"| grep -Eo "[0-9]{1,3}\.[0-9]{1,3}")
+  postfix=$(gcloud container images list-tags "${adapter_name}" --sort-by=~tags | grep " $version" |  grep -Eo "([0-9]{1,3}\.){2}[0-9]{1,3}" | head -1 | grep -Eo "[0-9]{1,3}" | tail -1)
+  postfix=$(($postfix+1))
+  gradle dockerBuildImage -Pdocker_tag=${adapter_name}:$version.$postfix
+  gradle dockerBuildImage -Pdocker_tag=${adapter_name}:latest
```

Comment:
We're going to avoid pushing to :latest for now to prevent users from depending on a version which could break. This  reduces the risk of new changes.

Reply:
we'll have to keep latest tag at build and tests steps, because this is default tag for image if tag is not specified, so i suggest to remove latest tag at publish step

Reply:
in cloudbuild yamls all variables resolved before any step is sterted
at step adapter-main we run freshly builded image, so we can't know exact version of it, if we left it empty then docker uses "latest" from local storage.
1 - we build image with tags: version, hash and latests
2 - tests
3 - we remove latests tag and publish image with tags: version, hash

Reply:
Instead of using the gcr.io/${_IMAGEPROJECT}/healthcare-api-dicom-dicomweb-adapter-import builder in the adapter-main step, can we instead run a bash command which runs the exact docker images we built in the build_adapter step? This would remove the problem of us picking up the :latest image from that gcr.io builder, if I understand correctly.

Reply:
>can we instead run a bash command which runs the exact docker images

it was my original idea but it won't work because when we run docker inside of docker images is not authorized with cloud build service account
so we have 2 workarounds here - run image from yaml or store service accout file in gcs, download it and pass into image at start

Reply:
If you build the docker image with a tag, you should be able to execute that tag as the builder. An example: https://cloud.google.com/cloud-build/docs/configuring-builds/build-test-deploy-artifacts#writing_build_requests_without_pushing_images

So we should be able to execute the local image by using its tag, right?

Reply:
>An example: https://cloud.google.com/cloud-build/docs/configuring-builds/build-test-deploy-artifacts#writing_build_requests_without_pushing_images
i'm actually doing exact same thing, just note what "- name: 'analyzer'" not using any 'analyzer', it's using 'analyzer:latest' if we won't specify tag, so we have :latest image into local storage because we know tag of image only on step of building it

Reply:
but i've forgot about commit hash, fixed

Reply:
Nice, thanks.